File: code/HuaWeiNote/app/src/main/java/com/fan/hwnote/app/model/entity/Folder.kt
package com.fan.hwnote.app.model.entity

/**
 * 文件夹实体（M12）。id == 0L 表示尚未持久化。
 *
 * 系统预置默认文件夹 id=1, isDefault=true，UI 不允许删除（仅允许改名）。
 */
data class Folder(
    val id: Long = 0L,
    val name: String,
    val orderIndex: Int = 0,
    val isDefault: Boolean = false,
    val deletedAt: Long = 0L,    // 0 = 未删除；>0 = 软删时间戳
)

File: code/HuaweiNoteFlutter/lib/models/folder.dart
class Folder {
  final int id;
  final String name;
  final int orderIndex;
  final bool isDefault;
  final int deletedAt;

  const Folder({
    required this.id,
    required this.name,
    this.orderIndex = 0,
    this.isDefault = false,
    this.deletedAt = 0,
  });
}
